forms.py

```diff
@@ -30,39 +30,41 @@ def save(self, commit=True):
         If there is already a matching object, it is returned
         rather than a new one to prevent duplicate bands from proliferating.
         """
-        band = super().save(commit=False)
+
+        # Get an abstract band for matching purposes
+        abstract_band = super().save(commit=False)
+        abstract_band.billing_agent = logic.determine_billing_agent(abstract_band.country)
+        abstract_band.fee, abstract_band.warnings = abstract_band.calculate_fee()
```

Comment:
The us of the name `abstract_band` made me read this more carefully, because the assigned object is a concrete instance of Band, there is no abstraction. I think unsaved_band or uncommitted_band is a more fitting name but with that out of the way it is clear what you are trying to do.

Reply:
Sure, I'm happy to make that change to `unsaved_band`.